railties/test/generators/app_generator_test.rb

```diff
@@ -1238,6 +1238,15 @@ def test_minimal_rails_app
     assert_no_gem "web-console", app_root
   end
 
+  def test_application_template_has_locale
+    app_root = File.join(destination_root, "myapp")
+    run_generator [app_root, "--minimal"]
+
+    assert_file "#{app_root}/app/views/layouts/application.html.erb" do |content|
+      assert_match(/I18n.locale/, content)
```

Comment:
Is this matching the string "I18n.locale"? Or should match the value, like "en"?

Reply:
This test doesn't check the outputted HTML, it checks that `I18n.locale` is inserted into the `application.html.erb` template. So the template would need to include `I18n.locale` (or maybe the full `lang=<%%= I18n.locale %>`), and in the rendered HTML that would evaluate to `lang="en"`. I'm not sure where to add a test that checks the rendered HTML, but this spec is only for the template.

Reply:
@metamoni Right, I was so confused but that makes sense 😅